server/sonar-server/src/main/java/org/sonar/server/component/DefaultRubyComponentService.java

```diff
@@ -71,20 +71,13 @@ public Long createComponent(String key, String name, String qualifier) {
 
   @CheckForNull
   public Long createComponent(String key, @Nullable String branch, String name, @Nullable String qualifier) {
-    // Sub view should not be created with provisioning. Will be fixed by http://jira.sonarsource.com/browse/VIEWS-296
-    if (!Qualifiers.SUBVIEW.equals(qualifier)) {
-      ComponentDto componentDto = componentService.create(NewComponent.create(key, name).setQualifier(qualifier).setBranch(branch));
-      if (componentDto == null) {
-        throw new BadRequestException(String.format("Component not created: %s", key));
-      }
-      ComponentDto component = (ComponentDto) resourceDao.selectByKey(componentDto.getKey());
-      if (component == null) {
-        throw new BadRequestException(String.format("Component not created: %s", key));
-      }
-      permissionService.applyDefaultPermissionTemplate(component.getKey());
-      return component.getId();
-    }
-    return null;
+    ComponentDto provisionedComponent = componentService.create(NewComponent.create(key, name).setQualifier(qualifier).setBranch(branch));
+    checkRequest(provisionedComponent != null, "Component not created: %s", key);
+    ComponentDto componentInDb = (ComponentDto) resourceDao.selectByKey(provisionedComponent.getKey());
+    checkRequest(componentInDb != null, "Component not created: %s", key);
+
+    permissionService.applyDefaultPermissionTemplate(componentInDb.getKey());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'componentInDb' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)